spring-xd-dirt/src/main/java/org/springframework/xd/dirt/integration/bus/RabbitBusCleaner.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.dirt.integration.bus;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpHost;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.AuthCache;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.auth.BasicScheme;
+import org.apache.http.impl.client.BasicAuthCache;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.protocol.HttpContext;
+
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+
+/**
+ *
+ * @author Gary Russell
+ */
+public class RabbitBusCleaner implements BusCleaner {
+
+	final static Log logger = LogFactory.getLog(RabbitBusCleaner.class);
+
+	@Override
+	public List<String> clean(String stream) {
+		return clean("http://localhost:15672", "guest", "guest", "/", "xdbus.", stream);
+	}
+
+	@SuppressWarnings("unchecked")
+	public List<String> clean(String adminUri, String user, String pw, String vhost,
+			String busPrefix, String stream) {
+		List<String> results = new ArrayList<>();
+		RestTemplate restTemplate = buildRestTemplate(adminUri, user, pw);
+		int n = 0;
+		while (true) {
+			String queueName = busPrefix + stream + "." + n++;
```

Comment:
Yes, the bus prefix is in the servers.yml so different container groups could use different queue prefixes. (Not sure why they would, but certainly possible - I'll add a utility method).